### DISCUSSION TITLE
Is it correct approach to create wrappers around useQuery fetch results ?

### DISCUSSION THREAD

**Discussion by WoodenPC** [TanStack/query#5328] [Q&A]
Hi. Lets suppose we have useQuery thats gets user data from api. User has some number of array fields. I want to pick fields from user data and use them in different components. However doing something like that everywhere seems to be wrong
```
const { data: user } = useQuery({
  queryFn: () => {
     // fetch user
   }
})

const someUserData = useMemo(() => {
 return user.someField.map(mapFn)
})
```

What i want to do is to create class named `UserModel `which has memoized method `userModel.getSomeUserData ` and inside `queryFn`
`return new UserModel(responseResult)`

Is that approach correct ? Maybe there are alternatives to avoid code duplication ?

**Comment by louis-young** [comment]
Hi 👋 

Please see this fantastic article discussing the different approaches: https://tkdodo.eu/blog/react-query-data-transformations
